pkg/tags/tag.go

```diff
@@ -280,3 +300,21 @@ func decodeInt64Splice(buffer *[]byte) int64 {
 	*buffer = (*buffer)[n:]
 	return val
 }
+
+// UpdateTag update the tag in the state store
+func (tag *Tag) UpdateTag() {
+	key := "tags_" + strconv.Itoa(int(tag.Uid))
+	value, err := tag.MarshalBinary()
+	if err != nil {
+		tag.logger.Warning("tag: ", err)
```

Comment:
why swallow the error here? it's better to return the error here so we know if there's a problem

Reply:
I agree, error here should be just returned, not logged. This function should return error and should be handler/propagated by the caller.